### PR TITLE
add feature `Chunk`

### PR DESCRIPTION
**Pull request by qianxi0410**
see: #1

### CODE REVIEW

File: chunk.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	for i := 0; i < n; i += size {
+		if i+size <= n {
+			tmp := make([]T, size)
+			copy(tmp, slice[i:i+size])
+			res = append(res, tmp)
+		} else {
+			tmp := make([]T, n-i)
+			copy(tmp, slice[i:])
+			res = append(res, tmp)
+			break
```
Comment (rakeeb-hossain) [inline chunk.go]: `break` is technically redundant here. If `i+size > n`, the next loop iteration will terminate anyway. Would prefer having this removed

File: chunk.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// returns a slice composed of multiple chunks of a user-specified size
+//
+// Vacuously, empty slices return empty regardless of the chunk size.
+//
+// When the chunk size is zero or negative, return an empty slice
+func Chunk[T any](slice []T, size int) [][]T {
+	if size <= 0 {
+		return [][]T{}
+	}
+	n := len(slice)
```
Comment (rakeeb-hossain) [inline chunk.go]: An edge case is when `n == 0`.  In the current logic, you return `{{}}` if `size > 0` and `{}` if `size <= 0`. I think it would make sense to return `{}` regardless of size if `n == 0` so I would modify this if-condition to `if size <= 0 || n == 0`. Please add a test case for this empty slice case too.

Comment (rakeeb-hossain) [reply chunk.go]: Happy to discuss this if you think the logic should be otherwise, btw!

File: chunk.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package functools
+
+import (
+	"math"
+)
+
+// Chunk consumes a slice of a generic type and
+// returns a slice composed of multiple chunks of a user-specified size
+//
+// Vacuously, empty slices return empty regardless of the chunk size.
```
Comment (rakeeb-hossain) [inline chunk.go]: Nit: change wording to `Vacuously, empty slices return an empty slice of slices regardless of the chunk size`

**Review by rakeeb-hossain — CHANGES_REQUESTED** [review]
Just some minor nits above regarding edges cases and wording, but otherwise looks great!

**Comment by rakeeb-hossain** [comment]
bump @qianxi0410

**Comment by qianxi0410** [comment]
Sorry, the reply was late due to the time difference. Your suggestion looks great, I have corrected it.
@rakeeb-hossain

**Comment by rakeeb-hossain** [comment]
LGTM!

**Review by rakeeb-hossain — APPROVED** [review]
